test/nodes/test_prompt_node.py

```diff
@@ -268,6 +268,37 @@ def test_complex_pipeline(prompt_model):
     assert "berlin" in result["results"][0].casefold()
 
 
+@pytest.mark.parametrize("prompt_model", ["hf", "openai"], indirect=True)
+@pytest.mark.parametrize("document_store", ["elasticsearch", "weaviate"], indirect=True)
+@pytest.mark.parametrize("retriever", ["bm25"], indirect=True)
```

Comment:
No need to use various retrievers and doc stores. Let's laser focus on the particular use case and ideally run this unit test only given documents directly passed to run method of the pipeline.

Reply:
@vblagoje, Sorry, but I can't pass the documents directly to the pipeline, as that would avoid triggering the bug this PR fixes.
The bug was, that when using the PromptNode with retrievers, the retrievers expected a `query` parameter - which is a string, while the PromptNode assumes that all its inputs are lists and handles them as lists without checking whether they are in fact lists.
If I provide the list of documents directly to the run method, then I am not testing the bug being fixed.

But if you want, I can set 1 retriever, 1 document store and 1 prompt model - so the scope of the test is decreased.

Reply:
I mean I can set the retriever to None and supply the documents directly to the Promptnode, but then I will have to write comments into the test explaining why we just don't change the `query` param to a list simply (because the `retriever` requires it as a string.
I am going to do that then.